e2e/src/config/jest/global-setup.ts

```diff
@@ -11,31 +11,41 @@ declare global {
 export default async (): Promise<void> => {
   const account = config.getL1AccountManager().whaleAccount(0);
   const l2Account = config.getL2AccountManager().whaleAccount(0);
-
-  const [dummyContract, l2DummyContract] = await Promise.all([
-    deployContract(new DummyContract__factory(), account),
-    deployContract(new DummyContract__factory(), l2Account),
-  ]);
-  console.log(`L1 Dummy contract deployed at address: ${await dummyContract.getAddress()}`);
-  console.log(`L2 Dummy contract deployed at address: ${await l2DummyContract.getAddress()}`);
-
-  const l2TestContract = await deployContract(new TestContract__factory(), l2Account);
-  console.log(`L2 Test contract deployed at address: ${await l2TestContract.getAddress()}`);
-
-  // Send ETH to the LineaRollup contract
   const lineaRollup = config.getLineaRollupContract(account);
   const l1JsonRpcProvider = config.getL1Provider();
 
-  const value = etherToWei("500");
+  const [l1AccountNonce, l2AccountNonce, { maxPriorityFeePerGas, maxFeePerGas }] = await Promise.all([
+    account.getNonce(),
+    l2Account.getNonce(),
+    l1JsonRpcProvider.getFeeData(),
+  ]);
+
   const fee = etherToWei("3");
   const to = "0x8D97689C9818892B700e27F316cc3E41e17fBeb9";
   const calldata = "0x";
-  const { maxPriorityFeePerGas, maxFeePerGas } = await l1JsonRpcProvider.getFeeData();
-  const tx = await lineaRollup.sendMessage(to, fee, calldata, { value, maxPriorityFeePerGas, maxFeePerGas });
-  await tx.wait();
+
+  const [dummyContract, l2DummyContract, l2TestContract] = await Promise.all([
+    deployContract(new DummyContract__factory(), account, [{ nonce: l1AccountNonce }]),
+    deployContract(new DummyContract__factory(), l2Account, [{ nonce: l2AccountNonce }]),
+    deployContract(new TestContract__factory(), l2Account, [{ nonce: l2AccountNonce + 1 }]),
+    // Send ETH to the LineaRollup contract
+    (
+      await lineaRollup.sendMessage(to, fee, calldata, {
+        value: etherToWei("500"),
+        maxPriorityFeePerGas,
+        maxFeePerGas,
+        nonce: l1AccountNonce + 1,
+      })
+    ).wait(),
+  ]);
+
+  console.log(`L1 Dummy contract deployed at address: ${await dummyContract.getAddress()}`);
+  console.log(`L2 Dummy contract deployed at address: ${await l2DummyContract.getAddress()}`);
+  console.log(`L2 Test contract deployed at address: ${await l2TestContract.getAddress()}`);
 
   console.log("Generating L2 traffic...");
-  const stopPolling = await sendTransactionsToGenerateTrafficWithInterval(l2Account, 2_000);
+  const pollingAccount = await config.getL2AccountManager().generateAccount(etherToWei("200"));
```

Comment:
Why not just use `l2Account` here but instead spending time to create a new account? Any particular reason?

Reply:
to avoid potential nonce issue with tests suites